readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -48,16 +50,8 @@ def __init__(self, *args, **kwargs):
                     self.project_path,
                     self.config_file,
                 )
-            self.old_artifact_path = os.path.join(
-                os.path.dirname(self.config_file),
-                self.sphinx_build_dir,
-            )
         except ProjectConfigurationError:
-            docs_dir = self.docs_dir()
-            self.old_artifact_path = os.path.join(
-                docs_dir,
-                self.sphinx_build_dir,
-            )
+            pass
```

Comment:
This definitely needs a comment.

Reply:
I may need to research a little more here to be able to add a comment 😄 . I deleted this code because `docs_dir` nor `self.old_artifact_path` were used anymore so I had to add a `pass`. However, I'm not sure what the try/except block is catching --maybe it can be removed as well? 🤔

Reply:
I added a big note explaining what's happening here. Note that _I'm not changing the current behavior_. However, I do not agree with the current behavior and I think we should refactor/change it to hard fail the build in of the cases and the other should probably managed differently (in general, I'd refactor the whole `Project.config_file` method and how it's used from outside).

All of this is outside the scope of this PR, tho, so I'm not making these changes here and I hope that the note I added to be enough for now.